### ISSUE TITLE
Improve spectrum weighting in table_psf_in_energy_band

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1740]
Triggered by https://github.com/PyCQA/pylint/issues/2410#issuecomment-415026690 I had a look at `table_psf_in_energy_band` and have some suggestions.

https://github.com/gammapy/gammapy/blob/cb993b580e8a54a776ed9fcaf534eddd320e79e4/gammapy/irf/psf_table.py#L520-L521

Currently there are two ways to define the spectrum to use for weighting. Pass a `spectrum` or a `spectral_index`. Passing a `spectrum` is completely untested, and given the complex way it's implemented with the min energy could be broken. See callers:

```
$ ack table_psf_in_energy_band gammapy
gammapy/irf/tests/test_psf_table.py
115:        psf_band = self.psf.table_psf_in_energy_band(energy_band)

gammapy/irf/psf_table.py
518:    def table_psf_in_energy_band(self, energy_band, spectral_index=2,

gammapy/data/tests/test_observations.py
113:    psf1_int = psf1.table_psf_in_energy_band(energy_band, spectral_index=2.3)
114:    psf2_int = psf2.table_psf_in_energy_band(energy_band, spectral_index=2.3)
117:    psf_tot_int = psf_tot.table_psf_in_energy_band(energy_band, spectral_index=2.3)
```

My suggestions would be to remove the `spectral_index` option, and to only keep the `spectrum=None` option and to make a `spectrum = PowerLaw(index=2.0)` as default, like @registerrier did here:
https://github.com/gammapy/gammapy/blob/ce4fe8a77d9bf8aded002234f0a50d136bdc63fa/gammapy/cube/exposure.py#L48

I'm not attempting this now, because the current implementation of `table_psf_in_energy_band` is pretty complex and has three TODO comments. I think it should be possible to compute the weights using Numpy array expressions, and also to apply them using a Numpy array expression, i.e. this should reduce to ~ 5 lines.

@registerrier or anyone - interested in this task?

**Comment by cdeil** [comment]
I'm moving this to v0.10 (January). @AtreyeeS or @registerrier - would you be willing to take care of improving this energy-weighted PSF implementation / tests? It's not labeled with "bug", but I'm not sure if it's correct or not.

**Comment by adonath** [comment]
Resolved #1975.
